src/main/java/org/opensearch/ad/transport/ADJobRunnerTransportAction.java

```diff
@@ -0,0 +1,120 @@
+        } catch (Exception e) {
+            LOG.error(e);
+            listener.onFailure(e);
+        }
+
+        try {
+            final AnomalyDetectorJob[] scheduledJobParameter = new AnomalyDetectorJob[1];
+            CompletableFuture<AnomalyDetectorJob[]> inProgressFuture = new CompletableFuture<>();
+
+            String jobParameterDocumentId = jobRunnerRequest.getJobParameterDocumentId();
+            if (jobParameterDocumentId == null || jobParameterDocumentId.isEmpty()) {
+                listener.onFailure(new IllegalArgumentException("jobParameterDocumentId cannot be empty or null"));
+            } else {
+                findById(jobParameterDocumentId, new ActionListener<>() {
+                    @Override
+                    public void onResponse(AnomalyDetectorJob anomalyDetectorJob) {
+                        scheduledJobParameter[0] = anomalyDetectorJob;
+                        inProgressFuture.complete(scheduledJobParameter);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.info("could not find AnomalyDetectorJob with id " + jobParameterDocumentId, e);
+                        inProgressFuture.completeExceptionally(e);
+                    }
+                });
+                JobExecutionContext jobExecutionContext = jobRunnerRequest.getJobExecutionContext();
+
+                AnomalyDetectorJobRunner.getJobRunnerInstance().runJob(scheduledJobParameter[0], jobExecutionContext);
+                JobRunnerResponse jobRunnerResponse = new JobRunnerResponse(true);
+                listener.onResponse(new ExtensionJobActionResponse<>(jobRunnerResponse));
+            }
+        } catch (Exception e) {
+            LOG.error(e);
+            listener.onFailure(e);
+        }
+
+    }
+
+    private void findById(String jobParameterId, ActionListener<AnomalyDetectorJob> listener) {
+        GetRequest getRequest = new GetRequest(AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX, jobParameterId);
+
+        try {
+            client.get(getRequest, ActionListener.wrap(response -> {
+                if (!response.isExists()) {
+                    listener.onResponse(null);
+                } else {
+                    try {
+                        XContentParser parser = XContentType.JSON
+                            .xContent()
+                            .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getSourceAsString());
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        listener.onResponse(AnomalyDetectorJob.parse(parser));
+                    } catch (IOException e) {
+                        logger.error("IOException occurred finding AnomalyDetectorJob for jobParameterId " + jobParameterId, e);
+                        listener.onResponse(null);
```

Comment:
why isn't this `listener.onFailure(e)`?

Reply:
ACK

Reply:
Uh, just reviewed but looking at unresolved comments this is still unchanged.

Reply:
In file changes the change is reflecting, here this file is outdated